### PR TITLE
Refactor UndoSnapshot

### PR DESCRIPTION
**Pull request by JiuqingSong**
This is to move function implementations out of UndoSnapshot class so that developer can easily build their own UndoSnapshot and reuse our existing logic. This can help the OWA code clean up and be able to put undo snapshot on store so that the code can easily clear undo snapshots without knowing Undo plugin.

### CODE REVIEW

File: packages/roosterjs-editor-core/lib/undo/UndoSnapshots.ts [hunk 1]
```diff
@@ -7,25 +14,20 @@ const MAXSIZELIMIT = 1e7;
...
  * A class to help manage undo snapshots
  */
-export default class UndoSnapshots implements UndoSnapshotsService {
-    private snapshots: string[];
-    private totalSize: number;
-    private currentIndex: number;
+export default class UndoSnapshots implements UndoSnapshotsService, Snapshots {
+    snapshots: string[] = [];
+    totalSize = 0;
+    currentIndex = -1;
```
Comment (titrongt) [inline packages/roosterjs-editor-core/lib/undo/UndoSnapshots.ts]: nit: Types were removed for `totalSize `and `currentIndex`

File: packages/roosterjs-editor-dom/lib/snapshots/clearSnapshotsAfterCurrent.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import canMoveCurrentSnapshot from './canMoveCurrentSnapshot';
+import { Snapshots } from 'roosterjs-editor-types';
+
+/**
+ * Clear all snapshots after the current one
+ * @param snapshots The snapshots data structure to clear
+ */
+export default function clearSnapshotsAfterCurrent(snapshots: Snapshots) {
```
Comment (titrongt) [inline packages/roosterjs-editor-dom/lib/snapshots/clearSnapshotsAfterCurrent.ts]: nit: the name `clearSnapshotsAfterCurrent` is a bit confusing to me, suggest change to something like `clearProceedingSnapshots`?

File: packages/roosterjs-editor-core/lib/undo/UndoSnapshots.ts (hunk 1 above)
Comment (JiuqingSong) [reply packages/roosterjs-editor-core/lib/undo/UndoSnapshots.ts]: yes. type is not necessary here

File: packages/roosterjs-editor-dom/lib/snapshots/clearSnapshotsAfterCurrent.ts (hunk 2 above)
Comment (JiuqingSong) [reply packages/roosterjs-editor-dom/lib/snapshots/clearSnapshotsAfterCurrent.ts]: sounds good.

**Pull request merged**